redbot/cogs/downloader/repo_manager.py

```diff
@@ -1119,28 +1122,58 @@ def get_all_cogs(self) -> Tuple[Installable, ...]:
         Tuple[Repo, Tuple[str, str]]
             A 2-`tuple` with Repo object and a 2-`tuple` of `str`
             containing old and new commit hashes.
-
         """
         repo = self._repos[repo_name]
         old, new = await repo.update()
         return (repo, (old, new))
 
-    async def update_all_repos(self) -> Dict[Repo, Tuple[str, str]]:
-        """Call `Repo.update` on all repositories.
+    async def update_repos(
+        self, repos: Optional[List[Repo]] = None
```

Comment:
This type hint is incorrect, the function accepts any iterable (and is even called with different iterables than list already):
```suggestion
        self, repos: Optional[Iterable[Repo]] = None
```
Remember to also update parameter type in docstring.